### PR TITLE
Enable test config for all pipeline test kinds

### PR DESCRIPTION
**Pull request by mtojek**
Resolves: https://github.com/elastic/elastic-package/issues/184

### CODE REVIEW

**Comment by mtojek** [comment]
PR to update the spec: https://github.com/elastic/package-spec/pull/88

File: docs/howto/pipeline_testing.md [hunk 1]
```diff
@@ -81,30 +102,12 @@ The pipeline test runner transforms log lines into file input events and sends t
 }
 ```
 
-The `multiline` section configures the log file reader to correctly detect multiline log entries using the `first_line_pattern`. Use this property if your logs may be split into multiple lines, e.g. Java stack traces.
+The `multiline` section (raw files only) configures the log file reader to correctly detect multiline log entries using the `first_line_pattern`. Use this property if your logs may be split into multiple lines, e.g. Java stack traces.
```
Comment (ycombinator) [inline docs/howto/pipeline_testing.md:105]: Maybe link `raw files` to the "Raw files" section above?

File: internal/testrunner/runners/pipeline/test_case.go [hunk 2]
```diff
@@ -21,22 +23,39 @@ type testCase struct {
...
+		if err != nil {
+			return nil, errors.Wrap(err, "marshalling mocked event failed")
+		}
+		events = append(events, m)
+	}
 	return &testCase{
 		name:   filename,
-		events: tcd.Events,
+		config: &config,
+		events: events,
```
Comment (ycombinator) [inline internal/testrunner/runners/pipeline/test_case.go:58]: With this change there is some duplication between the implementations of `createTestCaseForEvents` and `createTestCaseForRawInput` - essentially everything except the parsing of entries from either type of file and converting them into a standard `[]json.RawMessage` events.

I wonder if we should take this opportunity to reduce the responsibility of each of these functions to just loading the input data and converting it to standard `[]json.RawMessage` events and then pass the results to a common function that does the common processing with the test config and returns the `testCase` object?

File: docs/howto/pipeline_testing.md (hunk 1 above)
Comment (mtojek) [reply docs/howto/pipeline_testing.md]: Raw files linked.

File: internal/testrunner/runners/pipeline/test_case.go (hunk 2 above)
Comment (mtojek) [reply internal/testrunner/runners/pipeline/test_case.go]: Refactored

File: docs/howto/pipeline_testing.md [hunk 3]
```diff
@@ -81,30 +102,12 @@ The pipeline test runner transforms log lines into file input events and sends t
 }
 ```
 
-The `multiline` section configures the log file reader to correctly detect multiline log entries using the `first_line_pattern`. Use this property if your logs may be split into multiple lines, e.g. Java stack traces.
+The `multiline` section ([raw files](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md#raw-files) only) configures the log file reader to correctly detect multiline log entries using the `first_line_pattern`. Use this property if your logs may be split into multiple lines, e.g. Java stack traces.
```
Comment (ycombinator) [inline docs/howto/pipeline_testing.md:105]: Nit: I think you can just use a relative link here (just `#raw-files`).

**Review by ycombinator — APPROVED** [review]
LGTM.

File: docs/howto/pipeline_testing.md (hunk 3 above)
Comment (mtojek) [reply docs/howto/pipeline_testing.md]: Fixed, thanks!

**Pull request merged**